metrics/influxdb.go

```diff
@@ -0,0 +1,52 @@
+package metrics
+
+import (
+	"time"
+
+	gomet "github.com/rcrowley/go-metrics"
+	"github.com/vrischmann/go-metrics-influxdb"
+)
+
+// Influx :
+type Influx struct {
+	Addr, Username, Password string
+	Counter                  map[string]gomet.Counter
+	Histogram                map[string]gomet.Histogram
+}
+
+// New :
+func (influx Influx) New() Influx {
+	counter := make(map[string]gomet.Counter)
+	histrogram := make(map[string]gomet.Histogram)
+	counter[Requests] = gomet.NewCounter()
+	gomet.Register(Requests, counter[Requests])
+	counter[Successes] = gomet.NewCounter()
+	gomet.Register(Successes, counter[Successes])
+	sample := gomet.NewExpDecaySample(1028, 0.015)
+	histrogram[LatencyHistogram] = gomet.NewHistogram(sample)
+	gomet.Register(LatencyHistogram, histrogram[LatencyHistogram])
+	influx.Counter = counter
+	influx.Histogram = histrogram
+	return influx
+}
+
+// Monitor : implement Metrics
+func (influx Influx) Monitor(opts *ServerOpts) {
+	go influxdb.InfluxDB(
+		gomet.DefaultRegistry,
+		time.Second*5,
+		opts.Host,
+		opts.Database,
+		opts.Username,
+		opts.Password)
+}
+
+// CounterInc : implement Metrics
+func (influx Influx) CounterInc(name string) {
```

Comment:
This writes to influx on every counter inc?
Prometheus and Influx works very differently, and we don't want to write on every data change. I think we should try to batch the writes instead of sending a request each stat change